internal/server/handlers.go

```diff
@@ -262,25 +279,52 @@ func (h *handler) ResourceMarshalError(name string, resource proto.Message, err
 	}
 }
 
-func (h *handler) StartNotificationBatch() {
+func (h *handler) StartNotificationBatch(initialResourceVersions map[string]string) {
 	h.lock.Lock()
 	defer h.lock.Unlock()
 
+	// setInitialResourceVersion sets the initial version of resources to filter out unchanged resources.
+	if len(initialResourceVersions) > 0 {
+		h.initialResourceVersions = make(map[string]*initialResourceVersion, len(initialResourceVersions))
+		for name, version := range initialResourceVersions {
+			h.initialResourceVersions[name] = &initialResourceVersion{version: version}
+		}
+	}
 	h.batchStarted = true
 }
 
 func (h *handler) EndNotificationBatch() {
 	h.lock.Lock()
 	defer h.lock.Unlock()
 
+	h.handleDeletionsFromIRV()
 	h.batchStarted = false
-
 	if len(h.entries) > 0 {
 		h.immediateNotificationReceived.notify()
 		h.notificationReceived.notify()
 	}
 }
 
+func (h *handler) handleDeletionsFromIRV() {
+	for name, irv := range h.initialResourceVersions {
+		// if resource is not present on the server and client knows about it,
+		// that means resource is deleted and client don't know about it yet, (re-connect scenario) then sending resource deletion to client.
```

Comment:
Huge nit: can you avoid going over the 120 character line limit whenever possible? I have a shortcut to this action that automatically formats comments to be nice and aligned:
![image](https://github.com/user-attachments/assets/cd72fb2b-0896-451e-a61a-5eef052880cf)